File: aepc-angular-front/src/app/shared/models/employee/employee.model.ts
import { EmpRole } from './employee.role';
import { State } from './employee.state';

export class Employee {
  employeeId!: string;
  firstname!:string;
  lastname!: string;
  email!: string;
  hireDate!: Date;
  state!: State;
  role!: EmpRole;
  addressId!: string;
}

File: aepc-back-bs-ms-modules/aepc-api-spring-secur-oauth2-kafka-avro-bs-ms-employee/src/main/java/com/placide/k8skafkaavroaepccleanarchibsmicrosemployee/infra/adapters/output/models/EmployeeModel.java
package com.placide.k8skafkaavroaepccleanarchibsmicrosemployee.infra.adapters.output.models;

import com.placide.k8skafkaavroaepccleanarchibsmicrosemployee.infra.adapters.input.feignclients.models.AddressModel;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;


@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "employees")
public class EmployeeModel {
    @Id
    @GenericGenerator(name = "uuid")
    private String employeeId;
    private String firstname;
    private String lastname;
    @Column(unique = true)
    private String email;
    private String hireDate;
    private String state;
    private String role;
    private String addressId;
    @Transient
    private AddressModel address;
}
